src/ngraph/ngraph_emitter.cc

```diff
@@ -695,6 +695,42 @@ void Emitter::CreateBinaryOps() {
     return cast_result(CreateAutoBroadcast<ngraph::op::LessEq>(node),
                        getType(node->dtype_));
   };
+  ngraph_op_funcs_["smooth_l1"] = [this](const NodePtr& node) {
+    /* Smooth L1 Loss is a loss specific for R-CNN franchise training
+     * Smooth L1 Loss function:
+     * f(x) = 0.5 * (sigma * x) ^ 2,     |x| < 1 / sigma^2
+     *      = |x| - 0.5 / sigma / sigma, otherwise
+     * When sigma = 1, it is equivalent to the Huber loss, evaluated at
+     * delta = 1.
+     * smooth_l1_loss = w_out * f(w_in * x)
+     * with w_in, w_out provided by input_data.
+     */
+    auto input = op_map_[node->inputs_[0]];
+    auto sigma =
+        makeConstant(node, get_default(node, "scalar", std::string("0")));
+    auto sigma_sq = sigma * sigma;
+    auto inv_sigma_sq = std::make_shared<ngraph::op::Divide>(
+        makeConstant(node, "1.0"), sigma_sq);
+
+    // check if input is greater than inv_sigma_sq
+    auto is_input_gt_sigma =
+        std::make_shared<ngraph::op::Greater>(input, inv_sigma_sq);
+
+    // f(x) = 0.5 * (sigma * x) ^ 2
```

Comment:
💯 👍 🥇 for using comments to clarify what you intend to be computed by the nodes you're creating!

IMO it would be helpful for you to _not_ use the function name `f(x)` in this line and below, unless you're defining a function that specifically matches one of the two versions of `f(x)` that you describe in the block-comment at the top of this commit.

By using the same name `f(x)` in all of those places, it's unclear regarding which computations here and below are meant to exactly correspond fo the two definitions of `f(x)` you use in the block-comment above.